specs/trust-store-trust-policy.md

```diff
@@ -301,6 +344,51 @@ Signature verification levels provide defined behavior for each validation e.g.
     }
 ```
 
+#### Timestamp Countersignature Verification details
+
+Sufficient and necessary conditions to trigger timestamp countersignature verification:
+  - As the overall switch, `authenticTimestamp` under `signatureVerification` of trust policy is NOT marked as `skip`.
+  - The signing scheme is [`notary.x509`](./signing-scheme.md/#notaryx509).
+  - `tsa` trust store type is configured under `trustStores` field of the corresponding trust policy statement.
+  - `verifyTimestamp` under `signatureVerification` of trust policy is not set or set to `always`; OR it's set to `afterCertExpiry` and at least one certificate in the signing certificate chain has expired at the time of verification.
+
+See the flow chart below for a better understanding:
+```mermaid
+flowchart TD;
+  A{Authentic timestamp};
+  B{Signing scheme}; 
+  C{`tsa` trust store configured};
+  D{`verifyTimestamp`};
+  E{Signing cert chain expired};
+  R1[Trigger timestamp verification];
+  R2[No timestamp verification];
+
+  A --> |Not skip| B
+  A --> |Skip| R2
+  B --> |notary.x509| C
+  B --> |notary.x509.signingAuthority| R2
+  C --> |Yes| D
+  C --> |No| R2
+  D --> |Not set or `always`| R1
+  D --> |`afterCertExpiry`| E
+  E --> |Yes| R1
+  E --> |No| R2
+```
+
+Timestamp countersignature verification is a multi step process performs the following validations
+  - existence (timestamp token is present in the signature envelope)
+  - integrity (timestamp token is unaltered, countersignature is not corrupted)
+  - authenticity (the timestamp token is really from the TSA that claims to have signed it)
+  - revocation check (is the TSA still trusted at the time of verification)
+
+ **Existence** : Guarantees that the timestamp token exists in the signature envelope. It is always enforced when timestamp countersignature verification is triggered.
+
+ **Integrity** : Guarantees that the timestamp token wasn't altered after it was signed, or the countersignature isn't corrupted. It is always enforced when timestamp countersignature verification is triggered.
+
+ **Authenticity** : Guarantees that the timestamp was issued by a trusted TSA identity. Its definition does not include revocation, which is when a trusted TSA is subsequently untrusted because of a compromise. It is always enforced when timestamp countersignature verification is triggered.
```

Comment:
Shouldn't this be governed by [authenticity](https://github.com/notaryproject/specifications/blob/main/specs/trust-store-trust-policy.md#signature-verification-details) configuration in trust policy?

Reply:
Also included in issue #303.